### PR TITLE
 Build updates ported over from warehouse

### PR DESCRIPTION
**Pull request by yakloinsteak**
## Description

Ports over from the warehouse updates to the Dockerfiles to improve readability and build performance. Improves github action for better caching performance, vulnerability scans, and dockerfile linting.

## Type of change
- [x] Code clean-up / housekeeping

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)

### CODE REVIEW

**Comment by yakloinsteak** [comment]
@eanders This deploys and I can boot up things locally.

**Comment by yakloinsteak** [comment]
@eanders This deploys and I can boot up things locally. I think it's ready to get moved along.

File: bin/rails [hunk 1]
```diff
@@ -1,4 +1,27 @@
 #!/usr/bin/env ruby
-APP_PATH = File.expand_path('../config/application', __dir__)
-require_relative '../config/boot'
-require 'rails/commands'
```
Comment (eanders) [inline bin/rails:4]: Just noticing, but the warehouse still has bin stubs similar to this, so maybe we should apply whatever magic you did here over there as well?

File: docker/app/REVISION [hunk 2]
```diff
@@ -0,0 +1 @@
+d642734d6d5be5847222e429130cf2e586c81ee1
```
Comment (eanders) [inline docker/app/REVISION:1]: Should this have been committed?
It appears to be an empty file in the warehouse.

Comment (eanders) [reply docker/app/REVISION:1]: The seems to be working fine, system_status/details is reflecting different revisions for each deployment, so probably doesn't matter.

**Review by eanders — APPROVED** [review]
Thanks @yakloinsteak this looks great, and I _love_ that you removed the TOFU again.

File: docker/app/REVISION (hunk 2 above)
Comment (yakloinsteak) [reply docker/app/REVISION:1]: It just needs to exist so we could change the content to make it clearer it's just for testing

Comment (eanders) [reply docker/app/REVISION:1]: Thank you!

File: bin/rails (hunk 1 above)
Comment (yakloinsteak) [reply bin/rails:4]: I just made https://github.com/greenriver/hmis-warehouse/pull/4057 but I won't be able to test that out.

**Pull request merged**